WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt

```diff
@@ -833,6 +810,41 @@ class PostListViewModel @Inject constructor(
         _snackbarAction.postValue(snackbarHolder)
     }
 
+    fun scrollToPost(localPostId: Int) {
+        val data = pagedListData.value
+        if (data != null) {
+            _scrollToPosition.value = findItemListPosition(data, localPostId) ?: run {
+                AppLog.e(AppLog.T.POSTS, "ScrollToPost failed - the post not found.")
+                null
+            }
+        } else {
+            // store the target post id and scroll there when the data is loaded
+            scrollToLocalPostId = localPostId
+        }
+    }
+
+    fun onDataUpdated(data: PagedPostList) {
+        val localPostId = scrollToLocalPostId
+        if (localPostId != null) {
+            scrollToLocalPostId = null
+            _scrollToPosition.value = findItemListPosition(data, localPostId) ?: run {
```

Comment:
My comment for the `_scrollToPosition.value` in the `scrollToPost` applies here as well. We should also consider moving this to a helper. Maybe we could have a method that accepts both the `data` and the `localPostId` and scrolls to it. That'd also make it clearer what happens with `scrollToLocalPostId` variable. What do you think?

Reply:
👍